serverless/src/shared/getFilteredTriples.js

```diff
@@ -1,70 +1,109 @@
  * @param {string} params.version - The version of the concept scheme to query (e.g., 'published', 'draft', or a specific version number).
+ * @param {number} params.pageNum - The page number for pagination.
+ * @param {number} params.pageSize - The number of items per page.
  * @returns {Promise<Array>} A promise that resolves to an array of triple objects.
  *                           Each triple object contains 's' (subject), 'p' (predicate), and 'o' (object) properties.
  *
  * @throws {Error} If the HTTP request fails, if the response is not ok,
  *                 or if there's any error during the fetching or processing of the triples.
  *
  * @example
- * // Fetch triples for a specific concept scheme in the published version
+ * // Fetch triples for a specific concept scheme in the published version, page 1 with 10 items per page
  * try {
- *   const triples = await getFilteredTriples({ conceptScheme: 'sciencekeywords', version: 'published' });
+ *   const triples = await getFilteredTriples({
+ *     conceptScheme: 'sciencekeywords',
+ *     version: 'published',
+ *     pageNum: 1,
+ *     pageSize: 10
+ *   });
  *   console.log(triples);
  * } catch (error) {
  *   console.error('Failed to get triples:', error);
  * }
  *
  * @example
- * // Fetch triples matching a pattern in the draft version
+ * // Fetch triples matching a pattern in the draft version, page 2 with 20 items per page
  * try {
- *   const triples = await getFilteredTriples({ pattern: 'EARTH SCIENCE', version: 'draft' });
+ *   const triples = await getFilteredTriples({
+ *     pattern: 'EARTH SCIENCE',
+ *     version: 'draft',
+ *     pageNum: 2,
+ *     pageSize: 20
+ *   });
  *   console.log(triples);
  * } catch (error) {
  *   console.error('Failed to get triples:', error);
  * }
  *
- * @see getTriplesForConceptSchemeOrPatternQuery - Used to construct the SPARQL query.
- * @see sparqlRequest - Used to make the SPARQL query request.
+ * @see sparqlRequest - Used to make the SPARQL query requests.
+ * @see getConceptUrisQuery - Used to generate the SPARQL query for fetching concept URIs.
+ * @see getConceptDetailsQuery - Used to generate the SPARQL query for fetching concept details.
  */
 
-export const getFilteredTriples = async ({ conceptScheme, pattern, version }) => {
+export const getFilteredTriples = async ({
+  conceptScheme, pattern, version, pageNum, pageSize
+}) => {
+  const offset = (pageNum - 1) * pageSize
+
   try {
-    const response = await sparqlRequest({
+    // Step 1: Fetch concept URIs
+    const uriQuery = getConceptUrisQuery({
+      conceptScheme,
+      pattern,
+      pageSize,
+      offset
+    })
+    const uriResponse = await sparqlRequest({
+      method: 'POST',
+      contentType: 'application/sparql-query',
+      accept: 'application/sparql-results+json',
+      body: uriQuery,
+      version
+    })
+
+    if (!uriResponse.ok) {
+      throw new Error(`HTTP error! status: ${uriResponse.status}`)
```

Comment:
```suggestion
      throw new Error(`Failed to retrieve concept URIs ! status: ${uriResponse.status}`)
```
just to make it easier to debug later something that references the query that's failing in the log

Reply:
good suggestion.